apps/studio/src/lib/db/clients/mysql.ts

```diff
@@ -0,0 +1,1340 @@
+    } else {
+      config.ssl.rejectUnauthorized = server.config.sslRejectUnauthorized;
+    }
+  }
+
+  return config;
+}
+
+function resolveDefault(
+  version: MysqlClient["versionInfo"],
+  defaultValue: string
+) {
+  // adapted from https://github.com/PomeloFoundation/Pomelo.EntityFrameworkCore.MySql/pull/998/files
+  if (version.isMySql) return defaultValue;
+
+  if (!defaultValue) return null;
+
+  if (defaultValue.toString().toLowerCase() === "'null'") {
+    return null;
+  }
+
+  if (
+    defaultValue.startsWith("'") &&
+    defaultValue.endsWith("'") &&
+    defaultValue.length >= 2
+  ) {
+    // MariaDb escapes all single quotes with two single quotes in default value strings, even if they are
+    // escaped with backslashes in the original `CREATE TABLE` statement.
+    return defaultValue
+      .substring(1, defaultValue.length - 1)
+      .replaceAll("''", "'");
+  }
+
+  return defaultValue;
+}
+
+function identifyCommands(queryText: string) {
+  try {
+    return identify(queryText);
+  } catch (err) {
+    return [];
+  }
+}
+
+function isMultipleQuery(fields: any[]) {
+  if (!fields) {
+    return false;
+  }
+  if (!fields.length) {
+    return false;
+  }
+  return Array.isArray(fields[0]) || fields[0] === undefined;
+}
+
+function parseFields(fields: any[], rowsAsArray?: boolean) {
+  if (!fields) return [];
+  return fields.map((field, idx) => {
+    return { id: rowsAsArray ? `c${idx}` : field.name, ...field };
+  });
+}
+
+function parseRowQueryResult(
+  data: any,
+  rawFields: any[],
+  command: string,
+  rowsAsArray = false
+) {
+  // Fallback in case the identifier could not reconize the command
+  const fields = parseFields(rawFields, rowsAsArray);
+  const fieldIds = fields.map((f) => f.id);
+  const isSelect = Array.isArray(data);
+  return {
+    command: command || (isSelect && "SELECT"),
+    rows: isSelect
+      ? data.map((r: any) => (rowsAsArray ? _.zipObject(fieldIds, r) : r))
+      : [],
+    fields: fields,
+    rowCount: isSelect ? (data || []).length : undefined,
+    affectedRows: !isSelect ? data.affectedRows : undefined,
+  };
+}
+
+function filterDatabase(
+  item: Record<string, any>,
+  { database }: DatabaseFilterOptions = {},
+  databaseField: string
+) {
+  if (!database) {
+    return true;
+  }
+
+  const value = item[databaseField];
+  if (typeof database === "string") {
+    return database === value;
+  }
+
+  const { only, ignore } = database;
+
+  /* eslint-disable-next-line */
+  // @ts-ignore
+  if (only && only.length && !~only.indexOf(value)) {
+    return false;
+  }
+
+  /* eslint-disable-next-line */
+  // @ts-ignore
+  if (ignore && ignore.length && ~ignore.indexOf(value)) {
+    return false;
+  }
+
+  return true;
+}
+
+export class MysqlClient extends BasicDatabaseClient<ResultType> {
+  server: IDbConnectionServer;
+  database: IDbConnectionDatabase;
+  versionInfo: {
+    versionString: string;
+    version: number;
+    isMySql: boolean;
+    isMariaDb: boolean;
```

Comment:
If at all possible I think we should get rid of these. Any logic that isn't the same between the db's should be overridden in the child